src/event-log/event-stream.ts

```diff
@@ -0,0 +1,100 @@
+import type { MessageStore } from '../types/message-store.js';
+import type { EventsSubscribeMessage, EventsSubscription } from '../types/events-types.js';
+import type { EventStream, SubscriptionHandler } from '../types/subscriptions.js';
+import type { Filter, KeyValues } from '../types/query-types.js';
+import type { GenericMessage, GenericMessageSubscription } from '../types/message-types.js';
+
+import { EventEmitter } from 'events';
+import { EventsSubscribe } from '../interfaces/events-subscribe.js';
+import { EventsSubscriptionHandler } from '../handlers/events-subscribe.js';
+import { Message } from '../core/message.js';
+import { DwnError, DwnErrorCode } from '../core/dwn-error.js';
+
+const eventChannel = 'events';
+
+type EventStreamConfig = {
+  emitter?: EventEmitter;
+  reauthorizationTTL?: number;
+};
+
+export class EventStreamEmitter implements EventStream {
```

Comment:
EventStreamEmitter sounds like an Emitter of EventStreams (e.g. EventEmitter is and Emitter of Events).

Consider renaming to class to` EventStream` and rename the interface to `EventStreamInterface` or I am even onboard with `IEventStream` if you fine "Interface" too silly/verbose as a convention.